query/queryable.d.ts

```diff
@@ -70,21 +76,13 @@ export type Algebra = any;
 
 /**
  * Generic query engine interfaces.
- * It allow engines to return any type of result object for any type of query.
- * @param QueryFormatTypesAvailable The format of the query, either string or algebra object.
- * @param SourceType The allowed sources over which queries can be executed.
+ * It allow engines to return any type of result object for string queries.
  * @param SupportedMetadataType The allowed metadata types.
- * @param QueryType The allowed query types.
  * @param QueryStringContextType Type of the string-based query context.
- * @param QueryAlgebraContextType Type of the algebra-based query context.
  */
-export interface Queryable<
-  QueryFormatTypesAvailable extends string | Algebra,
-  SourceType,
+export interface StringQueryable<
   SupportedMetadataType,
-  QueryType extends Query<SupportedMetadataType>,
```

Comment:
This should remain, as engines won't be able to provide custom extensions of the query type otherwise. (Comunica requires this for instance)

Reply:
As per our conversation on Gitter, we agreed to postpone this to a later date if it turns out we really need it.